practices/libraries.md

```diff
@@ -0,0 +1,99 @@
+---
+slug: libraries
+stages:
+  - development
+short_description: Updating dependencies from your libraries can be a huge pain. But it is a necessary pain! Your whole project can be jeopardized without regular updates. Automatize and keep track of the dependencies’ changes.
+tags:
+  - libraries
+  - lockfile
+  - dependencies
+  - improvement
+  - functionality
+  - automated deployment
+  - continuous integration
+  - continuous delivery
+  - continuous deployment
+  - unit tests
+keywords:
+  - software library
+  - updating software
+  - updating dependencies
+  - updating libraries
+covered_by_dx_scanner: true
+---
+
+# Libraries
+
+**TL;DR**
+
+Updating dependencies from your libraries can be a huge pain. But it is a necessary pain! Your whole project can be jeopardized without regular updates. Automatize and keep track of the dependencies’ changes.
+
+## What Are the Libraries
+
+Libraries collect resources for software development. They consist of pre-written code, procedures, classes, scripts, data configurations, and more. Developers add libraries to the program they are developing to get some functionalities, dependencies, or automation without having to program it specifically for their program.
+
+Here are a few advantages in using libraries:
+
+- The library code can be used by multiple different programs. In comparison, the code that is part of the program can be used only within the program.
+- While using the library, the program gains the library’s dependencies and does not have to implement the dependencies itself.
```

Comment:
This is a weird sentence. While I use a library "A", it's quite common that the library "A" has its own dependencies e.g. on a library "B". So my app must install the library "A" and "B" too and it's not about the implementation because the library "A" has its own functionality built on library "B".

I'll explain it to you tomorrow.

Reply:
`While using the library you get the code you don't have to program yourself.`